### PR TITLE
Fix App `Keypoint` filters

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

Fixes App `Keypoint` filters

```python
import numpy as np

import fiftyone as fo
import fiftyone.zoo as foz
from fiftyone import ViewField as F

dataset = foz.load_zoo_dataset("coco-2017", split="validation").clone()

model = foz.load_zoo_model("keypoint-rcnn-resnet50-fpn-coco-torch")
dataset.default_skeleton = model.skeleton

view = (
    dataset
    .match_labels(fields="ground_truth", filter=F("label") == "person")
    .limit(2)
)

view.apply_model(model, label_field="predictions")

# Add some custom attributes
for sample in view:
    for keypoint in sample["predictions_keypoints"].keypoints:
        keypoint["occluded"] = [np.random.randn() > 0 for _ in range(len(keypoint.points))]
        keypoint["age"] = np.random.randint(51)

    sample.save()
``` 

### Before

https://github.com/voxel51/fiftyone/assets/19821840/3d62c400-4e0c-4b76-bcbb-55c70103c16a

### After

https://github.com/voxel51/fiftyone/assets/19821840/b29dc19b-9973-47f1-a369-927c876ff04d

## How is this patch tested? If it is not, please explain why.

Recoil unit test

## Release Notes

* Fixed `Keypoint` filtering in the sidebar

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **Refactor**
  - Improved error handling for missing mock values in the app's mock value store.
  - Streamlined count calculation in the String Filter component for enhanced code organization.
  - Updated the retrieval method for `aggCount` to ensure accurate count calculations.
  - Modified the identification key for the `isListField` selector to improve selector specificity.

- **New Features**
  - Introduced a new function for dynamic count determination in the String Filter component.

- **Tests**
  - Added new test cases for verifying counts related to specific paths using Recoil selectors.

### CODE REVIEW

File: app/__mocks__/recoil.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export function setMockAtoms(newMockValues: { [key: string]: any }) {
 export const getValue = (atom) => {
   if (mockValuesStore[atom.key]) {
     const str = JSON.stringify(atom.params);
-    if (mockValuesStore[atom.key].hasOwnProperty(str)) {
+    if (mockValuesStore[atom.key].hasOwn(str)) {
```
Comment (imanjra) [inline app/__mocks__/recoil.ts:18]: Is `hasOwn` a thing? I could be wrong but I remember something like `Object.hasOwn(some_object,'some_key')`

File: app/packages/state/src/recoil/pathData/counts.test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { describe, expect, it, vi } from "vitest";
```
Comment (imanjra) [inline app/packages/state/src/recoil/pathData/counts.test.ts:1]: 🧪

**Review by imanjra — APPROVED** [review]
Left a minor comment but tested locally and LGTM

File: app/__mocks__/recoil.ts (hunk 1 above)
Comment (benjaminpkane) [reply app/__mocks__/recoil.ts]: Fixed! Thank you 🙏

**Pull request merged**